e2e/nomostest/taskgroup/taskgroup.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taskgroup
+
+import (
+	"sync"
+
+	"go.uber.org/multierr"
+)
+
+// Task is a function that returns an error.
+// See `taskgroup.New`.
+type Task func() error
+
+// Filter is a function that takes an error and returns it (or a modified error)
+// if it should be retained by `TaskGroup.Wait`.
+type Filter func(error) error
+
+// ErrorGroup is an error that contains a list of errors.
+// This interface is implimented by `multierr.multiError`
+type ErrorGroup interface {
+	Errors() []error
+	Error() string
+}
+
+// TaskGroup is a helper to make it easy to execute one or more `Task` functions
+// and collect their errors, if any.
+type TaskGroup struct {
+	errCh  chan error
+	wg     sync.WaitGroup
+	filter Filter
+}
+
+// New returns a new TaskGroup, which can be used to execute one or more `Task`
+// functions and collect their errors, if any.
+func New() *TaskGroup {
+	return &TaskGroup{
+		errCh: make(chan error),
+	}
+}
+
+// NewWithFilter returns a new TaskGroup, like `New`, but with a filter for
+// optionally ignoring/muting errors using a `Filter` function.
+func NewWithFilter(f Filter) *TaskGroup {
```

Comment:
I'm thinking a variable arg for the Filter would offer more flexibility here. That way we could have some narrowly scoped Filter functions that can be reused rather than having to create a new Filter function for combinations of errors to ignore. This would also give the option of a single `New` function that serves all purposes (no Filter, single Filter, multiple Filters, etc). WDYT?

Reply:
Don't have any use cases yet, so i'm not inclined to add them yet. 

I copied the idea from https://github.com/creachadair/taskgroup because it was quick and easy, but my impl waits until all are complete and batches the errors in a MultiError, instead of just returning the first error.

Reply:
meh... i'll add it. what the hell

Reply:
lol too late